### PR TITLE
Update to ethers v6

### PR DESCRIPTION
**Pull request by compojoom**
Not following the PR guide as this PR is neither a feature, nor a bug fix. So neither of the templates was fiting. 

This PR updates ethers.js dependency to the currently latest ethers-v6 - 6.9.2

### Implementation
All tests have been adjusted and pass.
Typescript definitions have also been updated. 

### Additional Context
We need this because we are upgrading the safe-wallet-app to use ethers-v6 (https://github.com/safe-global/safe-wallet-web/pull/3087) and having zodiac rely on ethers-v5 screws with our types and requires us to do some gymnastics around types... Also we end up with BigNumber and bigints and we would just want to move ahead :)

### CODE REVIEW

**Comment by compojoom** [comment]
I have read the CLA Document and I hereby sign the CLA

File: contracts/Imports.sol
Comment (jfschwarz) [inline contracts/Imports.sol:1]: Why is this file required?

Comment (jfschwarz) [inline contracts/Imports.sol:1]: Why is this file required?

**Review by jfschwarz — CHANGES_REQUESTED** [review]
Thank you for this! 🙏

File: contracts/Imports.sol
Comment (compojoom) [reply contracts/Imports.sol:1]: the sdk tests were failing as they couldn't find the mock contract, but it must have been a different issue as they now work even would that import.

**Review by jfschwarz — APPROVED** [review]
lgtm 🚀

**Pull request merged**